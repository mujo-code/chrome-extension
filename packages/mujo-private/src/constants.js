// config
export const API_HOST = 'https://getmujo.com'

// storage
export const SITE_TIME_KEY = 'SITE_TIME'
export const PREDICTED_BREAK_TIMES = 'PREDICTED_BREAK_TIMES'

// features
export const SMART_BREAKS_FEATURE = true

// alarms
export const SMART_BREAK = 'SMRT_BRK' // is only part of the key
// ex. SMRT_BRK_2019-08-07T22:17:13.246Z
