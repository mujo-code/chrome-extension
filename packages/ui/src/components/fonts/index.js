import { Global, css } from '@emotion/core'
import { omitKeys } from '@mujo/box'
import React from 'react'
import { Box } from '../box'

const sup = { ' sup': { fontSize: '0.6em' } }

export const Font = props => (
  <Global
    styles={css`
      #mujo-extension *,
      .mujo-modal * {
        font-family: 'IBM Plex Sans', sans-serif;
        --webkit-font-smoothing: antialiased;
        --moz-osx-font-smoothing: grayscale;
      }
    `}
  />
)

const FontBox = props => {
  const otherProps = omitKeys(
    props,
    'Component',
    'color',
    'marginTop',
    'marginBottom',
    'css'
  )
  const overrides = Array.isArray(props.css) ? props.css : [props.css]
  return (
    <Box
      Component={props.Component || 'p'}
      color={props.color || 'color'}
      marginTop={props.marginTop || 'm'}
      marginBottom={props.marginBottom || 'm'}
      css={[sup, ...overrides]}
      {...otherProps}
    />
  )
}

export const HeaderL = props => (
  <FontBox Component="h2" font="headerL" {...props} />
)

export const HeaderS = props => (
  <FontBox
    Component="h4"
    marginTop="s"
    marginBottom="s"
    font="headerS"
    {...props}
  />
)

export const BodyL = props => <FontBox font="bodyL" {...props} />

export const BodyS = props => <FontBox font="bodyS" {...props} />

export const Link = props => (
  <FontBox
    Component="a"
    color="danube"
    cursor="pointer"
    textDecoration="underline"
    {...props}
  />
)

export const FixedL = props => (
  <FontBox Component="span" font="fixedL" {...props} />
)
export const FixedS = props => (
  <FontBox Component="span" font="fixedS" {...props} />
)

export const Span = props => (
  <FontBox
    Component="span"
    marginTop={props.marginTop || 'zero'}
    marginBottom={props.marginTop || 'zero'}
    {...props}
  />
)

export const Sup = props => (
  <FontBox
    Component="sup"
    marginTop={props.marginTop || 'zero'}
    marginBottom={props.marginTop || 'zero'}
    {...props}
  />
)
