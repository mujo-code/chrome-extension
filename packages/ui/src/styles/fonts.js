const medium = 700
const extraLight = 200

const fontWeight = {
  medium: { fontWeight: medium },
  extraLight: { fontWeight: extraLight },
}

export const fontStyles = {
  fontWeight,
  font: {
    headerL: {
      fontWeight: medium,
      fontSize: 32,
      lineHeight: '40px',
    },

    headerS: {
      fontWeight: medium,
      fontSize: 16,
      lineHeight: '24px',
    },

    bodyXl: {
      fontWeight: extraLight,
      fontSize: 32,
      lineHeight: '40px',
    },

    bodyL: {
      fontWeight: extraLight,
      fontSize: 24,
      lineHeight: '32px',
    },

    bodyS: {
      fontWeight: extraLight,
      fontSize: 16,
      lineHeight: '24px',
    },

    fixedL: {
      fontFamily: '"Monaco", monospace',
      fontSize: 24,
      lineHeight: '32px',
    },

    fixedS: {
      fontFamily: '"Monaco", monospace',
      fontSize: 16,
      lineHeight: '24px',
    },
  },
}
