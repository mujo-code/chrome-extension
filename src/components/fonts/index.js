import { Box } from '@jcblw/box'
import React from 'react'
import {
  headerL,
  headerS,
  bodyL,
  bodyS,
  fixedL,
  fixedS,
} from './styles'

const additionalStyles = { ' sup': { fontSize: '0.6em' } }

const FontBox = props => (
  <Box
    Component={props.Component || 'p'}
    color={props.color || 'color'}
    marginTop={props.marginTop || 'm'}
    marginBottom={props.marginBottom || 'm'}
    css={additionalStyles}
    {...props}
  />
)

export const HeaderL = props => (
  <FontBox Component="h2" {...headerL} {...props} />
)

export const HeaderS = props => (
  <FontBox
    Component="h4"
    marginTop="s"
    marginBottom="s"
    {...headerS}
    {...props}
  />
)

export const BodyL = props => <FontBox {...bodyL} {...props} />

export const BodyS = props => <FontBox {...bodyS} {...props} />

export const Link = props => (
  <FontBox
    Component="a"
    color="danube"
    cursor="pointer"
    textDecoration="underline"
    {...props}
  />
)

export const FixedL = props => (
  <FontBox Component="span" {...fixedL} {...props} />
)
export const FixedS = props => (
  <FontBox Component="span" {...fixedS} {...props} />
)

export const Span = props => (
  <FontBox
    Component="span"
    marginTop={props.marginTop || 'zero'}
    marginBottom={props.marginTop || 'zero'}
    {...props}
  />
)

export const Sup = props => (
  <FontBox
    Component="sup"
    marginTop={props.marginTop || 'zero'}
    marginBottom={props.marginTop || 'zero'}
    {...props}
  />
)
