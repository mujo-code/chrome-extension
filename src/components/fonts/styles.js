import { css } from 'glamor'

css.global('#mujo-extension *, .mujo-modal *', {
  fontFamily: "'IBM Plex Sans', sans-serif",
  WebkitFontSmoothing: 'antialiased',
  MozOsxFontSmoothing: 'grayscale',
})

const medium = { fontWeight: 700 }

const extraLight = { fontWeight: 200 }

export const headerL = css(medium, {
  fontSize: 32,
  lineHeight: '40px',
})

export const headerS = css(medium, {
  fontSize: 16,
  lineHeight: '24px',
})

export const bodyXl = css(extraLight, {
  fontSize: 32,
  lineHeight: '40px',
})

export const bodyL = css(extraLight, {
  fontSize: 24,
  lineHeight: '32px',
})

export const bodyS = css(extraLight, {
  fontSize: 16,
  lineHeight: '24px',
})

export const fixedL = css(medium, {
  fontFamily: '"Monaco", monospace',
  fontSize: 24,
  lineHeight: '32px',
})

export const fixedS = css(medium, {
  fontFamily: '"Monaco", monospace',
  fontSize: 16,
  lineHeight: '24px',
})

export const underline = css({ textDecoration: 'underline' })
