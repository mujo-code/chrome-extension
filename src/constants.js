// Config values
export const MINUTE = 1000 * 60
export const FIVE_MINUTES = MINUTE * 5
export const FOURTY_FIVE_MINUTES = MINUTE * 45
export const HOUR = MINUTE * 60
export const THREE_HOURS = HOUR * 3
export const ALARM_DEFAULT_VALUE = true
export const MAX_BREAKTIMERS = 5
export const CURRENT_SUB_SKU = 'early_subscription'
export const APP_READY_KEY = '__mujo_ready__'

// ALARMS
export const ALARM_KEY = 'MINDFUL_ALARM' // also storage key
export const HEARTBEAT = 'HEARTBEAT'

// DB Keys
export const DATABASE_NAME = 'MUJO'
export const DATABASE_STORE = 'STATS'
export const DATABASE_VERSION = 1
export const LAST_ACTIVITY_TABLE = 'LAST_ACTIVITY'

// Storage keys
export const TOP_SITES_KEY = 'TOP_SITES'
export const TOP_SITES_USAGE_KEY = 'TOP_SITES_USAGE'
export const PAGE_VIEWS_KEY = 'PAGE_VIEWS'
export const SHOW_TOP_SITES_KEY = 'SHOW_TOP_SITES'
export const SITE_TIME_KEY = 'SITE_TIME'
export const BREAK_TIMERS_KEY = 'BREAK_TIMERS'
export const LAST_ACTIVE_KEY = 'LAST_ACTIVE_KEY'
export const ACTIVITY_NUMBER_KEY = 'ACTIVITY_NUMBER_KEY'

// Actions
export const NEW_TAB_CONNECTION = 'New Connection'
export const PAGE_VIEWING_TIME = 'Page Viewing Time'
export const DEEP_LINK_NEWTAB = 'Deep Link New Tab'
export const RESET_USAGE = 'Reset Usage Stats'
export const TRACK = 'Track'
export const ADD_BROADCAST_TAB = 'Add Broadcast Tab'
export const REMOVE_BROADCAST_TAB = 'Remove Broadcast Tab'
export const VALUE_CHANGED = 'Value changed'

// Getter, Setters for storage
export const GET_STORAGE = 'GET_STORAGE'
export const SET_STORAGE = 'SET_STORAGE'

// Feature Flags
export const SCREEN_TIME_FEATURE = true
export const BREAK_TIMER_FEATURE = true
export const SUBSCRIBE_FEATURE = false

// Upsell modals
export const MAX_BREAKTIMER_MODAL = 'breakTimerMax'
export const SUB_DETAILS_MODAL = 'subscriptionDetails'
export const SUB_SUCCESS_MODAL = 'subSuccess'

// OPTIONAL permissions requested
export const SCREEN_TIME_PERMISSIONS = {
  permissions: ['activeTab'],
  origins: ['https://*/*', 'http://*/*'],
}
