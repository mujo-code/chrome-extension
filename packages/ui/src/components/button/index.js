import { omitKeys } from '@mujo/box'
import { Tracker } from '@mujo/utils'
import React, { useState } from 'react'
import { Box } from '../box'
import { headerS } from '../fonts/styles'
import { ToolTip } from '../tool-tip'

const { track } = Tracker

const colors = {
  primary: {
    color: 'mischka',
    backgroundColor: 'outerSpace',
    highlight: 'saltBox',
  },
  secondary: {
    color: 'white',
    backgroundColor: 'saltBox',
    highlight: 'gravel',
  },
  tertiary: {
    color: 'outerSpace',
    backgroundColor: 'mischka',
    highlight: 'saltBox',
  },
}

export const Button = props => {
  const { children = 'Label', design = 'primary', alt, altOffset = 0 } = props
  const restProps = omitKeys(
    props,
    'design',
    'alt',
    'children',
    'altOffset',
    'onClick'
  )
  const [tooltipOpen, setToolTipOpen] = useState(false)
  const { color, backgroundColor, highlight } = colors[design]
  return (
    <Box
      font="headerS"
      outlineColor={highlight}
      color={color}
      backgroundColor={backgroundColor}
      display="inlineFlex"
      flexDirection="column"
      alignItems="center"
      paddingTop="xs"
      paddingLeft="l"
      paddingBottom="xs"
      paddingRight="l"
      border="none"
      borderRadius="l"
      textDecoration="none"
      position="relative"
      onClick={(e, ...args) => {
        props.onClick && props.onClick(e, ...args)
        track({
          category: 'button',
          action: 'click',
          label: typeof children === 'string' ? children : e.target.textContent,
        })
      }}
      onMouseLeave={() => setToolTipOpen(false)}
      onMouseEnter={() => setToolTipOpen(true)}
      {...restProps}
    >
      {children}
      {alt ? (
        <ToolTip isOpen={tooltipOpen} offset={altOffset}>
          {alt}
        </ToolTip>
      ) : null}
    </Box>
  )
}

Button.defaultProps = { Component: 'button' }
