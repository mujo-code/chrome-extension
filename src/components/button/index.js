import { Box } from '@jcblw/box'
import { removeKeys } from '@jcblw/box/dist/lib/remove-keys'
import React, { useState } from 'react'
import { headerS } from '../fonts/styles'
import { ToolTip } from '../tool-tip'

const colors = {
  primary: {
    color: 'mischka',
    backgroundColor: 'outerSpace',
    highlight: 'saltBox',
  },
  secondary: {
    color: 'white',
    backgroundColor: 'saltBox',
    highlight: 'gravel',
  },
  tertiary: {
    color: 'outerSpace',
    backgroundColor: 'mischka',
    highlight: 'saltBox',
  },
}

export const Button = props => {
  const {
    children = 'Label',
    design = 'primary',
    alt,
    altOffset = 0,
  } = props
  const restProps = removeKeys(
    props,
    'design',
    'alt',
    'children',
    'altOffset'
  )
  const [tooltipOpen, setToolTipOpen] = useState(false)
  const { color, backgroundColor, highlight } = colors[design]
  return (
    <Box
      outlineColor={highlight}
      color={color}
      backgroundColor={backgroundColor}
      display="inlineFlex"
      direction="column"
      alignItems="center"
      paddingTop="xs"
      paddingLeft="l"
      paddingBottom="xs"
      paddingRight="l"
      border="none"
      borderRadius="l"
      textDecoration="none"
      position="relative"
      onMouseLeave={() => setToolTipOpen(false)}
      onMouseEnter={() => setToolTipOpen(true)}
      {...headerS}
      {...restProps}
    >
      {children}
      {alt ? (
        <ToolTip isOpen={tooltipOpen} offset={altOffset}>
          {alt}
        </ToolTip>
      ) : null}
    </Box>
  )
}

Button.defaultProps = { Component: 'button' }
