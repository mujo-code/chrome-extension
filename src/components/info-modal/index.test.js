import React from 'react'
import { create } from 'react-test-renderer'
import { MAX_BREAKTIMER_MODAL } from '../../constants'
import { Button } from '../button'
import { HeaderL, BodyL } from '../fonts'
import { InfoModal } from '.'

test('InfoModal should match snapshot', () => {
  // Use break timer modal for snapshot
  const context = { name: MAX_BREAKTIMER_MODAL, url: 'foo' }
  const tree = create(
    <InfoModal isOpen={true} context={context} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('InfoModal should allow for custom contexts', () => {
  const context = { title: 'foo', description: ['bar'] }
  const { root } = create(
    <InfoModal isOpen={true} context={context} />
  )
  expect(root.findByType(HeaderL).props.children).toBe('foo')
  expect(root.findByType(BodyL).props.children).toBe('bar')
})

test('InfoModal should allow for clicks to buttons', done => {
  expect.assertions(2)
  const context = {
    button: {
      children: 'foo',
      onClick: () => {
        expect(true).toBe(true)
        done()
      },
    },
  }
  const { root } = create(
    <InfoModal isOpen={true} context={context} />
  )
  const button = root.findByType(Button)
  button.props.onClick()
  expect(button.props.children).toBe('foo')
})
