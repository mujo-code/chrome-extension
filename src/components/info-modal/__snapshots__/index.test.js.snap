// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoModal should match snapshot 1`] = `
<div
  className="css-ehyxbj"
  closeTimeoutMS={200}
  data-mujo={true}
  isOpen={true}
  outlineColor="saltBox"
  overlayClassName="mujo-modal css-qcfpiy"
  portalClassName="css-1bdqre4"
  shouldCloseOnOverlayClick={true}
>
  <div
    className="css-1mv301s"
    position="absolute"
  >
    <svg
      fill="none"
      height={32}
      icon="close"
      stroke="#EAE2EB"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width={32}
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="18"
        x2="6"
        y1="6"
        y2="18"
      />
      <line
        x1="6"
        x2="18"
        y1="6"
        y2="18"
      />
    </svg>
  </div>
  <div
    className="css-rczwh0"
  >
    <h2
      className="css-1lzekp8"
      color="color"
    >
      Mindful to the max
    </h2>
  </div>
  <div
    className="css-1yhq37y"
    overflow="scroll"
  >
    <p
      className="css-1x9rsfv"
      color="color"
    >
      You have set the max number of break timers. To get access to unlimited break timers and add a break timer for foo subscribe to Muj≈ç.
    </p>
  </div>
  <div
    backgroundColor="gravel"
    className="css-1yqdfgl"
  >
    <button
      backgroundColor="saltBox"
      className="css-15foh21"
      color="white"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      outlineColor="gravel"
      position="relative"
    >
      Subscription Details
    </button>
  </div>
</div>
`;
